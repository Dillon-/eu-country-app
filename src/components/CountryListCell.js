import React from 'react';
import { View, Text } from 'react-native';
import PropTypes from 'prop-types';
import EStyleSheet from 'react-native-extended-stylesheet-forked';

const CountryListCell = ({ country: { name, subregion, capital, languages } }) => {
  return (
    <View style={styles.container}>
      <Text style={[styles.text, styles.headerText]}>{name}</Text>
      <Text style={styles.text}>{subregion}</Text>
      <Text style={styles.text}>Capital: {capital}</Text>
      <Text style={styles.text}>Lang: {languages.map(language => language.name).join(', ')}</Text>
    </View>
  );
};

CountryListCell.propTypes = {
  country: PropTypes.shape({
    name: PropTypes.string.isRequired,
    subregion: PropTypes.string.isRequired,
    capital: PropTypes.string.isRequired,
    languages: PropTypes.array.isRequired,
  }),
};

const styles = EStyleSheet.create({
  container: {
    backgroundColor: '$lightGray',
    marginHorizontal: 20,
    marginVertical: 10,
    padding: 10,
  },
  headerText: {
    fontSize: 22,
    fontWeight: '800',
  },
  text: {
    color: '$black',
    fontSize: 16,
  }
});

export default CountryListCell;
