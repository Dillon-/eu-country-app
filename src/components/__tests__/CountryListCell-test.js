import React from 'react';
import renderer from 'react-test-renderer';
import CountryListCell from '../CountryListCell';

it('renders a CountryListCell', () => {

  const country = {

    name: 'Sweden',
    subregion: 'Northern Europe',
    capital: 'Stockholm',
    languages: [{ name: 'Swedish' }]
  }
  const component = renderer.create(
    <CountryListCell
      country={country}
    />
  );
  expect(component).toMatchSnapshot();
});