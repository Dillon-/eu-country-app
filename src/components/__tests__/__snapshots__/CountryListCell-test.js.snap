// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a CountryListCell 1`] = `
<View
  style={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        undefined,
        undefined,
      ]
    }
  >
    Sweden
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={undefined}
  >
    Northern Europe
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={undefined}
  >
    Stockholm
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={undefined}
  >
    Lang: 
    Swedish
  </Text>
</View>
`;
